src/main/java/eu/dissco/core/datacitepublisher/service/DataCitePublisherService.java

```diff
@@ -0,0 +1,311 @@
+import eu.dissco.core.datacitepublisher.domain.datacite.DcCreator;
+import eu.dissco.core.datacitepublisher.domain.datacite.DcData;
+import eu.dissco.core.datacitepublisher.domain.datacite.DcDate;
+import eu.dissco.core.datacitepublisher.domain.datacite.DcDescription;
+import eu.dissco.core.datacitepublisher.domain.datacite.DcNameIdentifiers;
+import eu.dissco.core.datacitepublisher.domain.datacite.DcRelatedIdentifiers;
+import eu.dissco.core.datacitepublisher.domain.datacite.DcRequest;
+import eu.dissco.core.datacitepublisher.domain.datacite.DcSubject;
+import eu.dissco.core.datacitepublisher.domain.datacite.DcTitle;
+import eu.dissco.core.datacitepublisher.domain.datacite.DcType;
+import eu.dissco.core.datacitepublisher.domain.datacite.UriScheme;
+import eu.dissco.core.datacitepublisher.exceptions.DataCiteApiException;
+import eu.dissco.core.datacitepublisher.exceptions.DataCiteMappingException;
+import eu.dissco.core.datacitepublisher.exceptions.InvalidFdoProfileRecievedException;
+import eu.dissco.core.datacitepublisher.kafka.KafkaPublisherService;
+import eu.dissco.core.datacitepublisher.schemas.DigitalSpecimen;
+import eu.dissco.core.datacitepublisher.schemas.MediaObject;
+import eu.dissco.core.datacitepublisher.component.XmlLocReader;
+import eu.dissco.core.datacitepublisher.web.DataCiteClient;
+import java.time.DateTimeException;
+import java.time.ZonedDateTime;
+import java.util.ArrayList;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.http.HttpMethod;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+@Slf4j
+public class DataCitePublisherService {
+
+  private final KafkaPublisherService kafkaPublisherService;
+  private final XmlLocReader xmlLocReader;
+  @Qualifier("objectMapper")
+  private final ObjectMapper mapper;
+  private final DataCiteClient dataCiteClient;
+
+  public void handleMessages(DigitalSpecimenEvent digitalSpecimenEvent) {
+    var dcRequests = digitalSpecimenEvent.pidRecords().stream().map(this::buildDcRequest)
+        .toList();
+    publishToDataCite(dcRequests, digitalSpecimenEvent.eventType());
+  }
+
+  public void handleMessages(MediaObjectEvent mediaObjectEvent) {
+    var dcRequests = mediaObjectEvent.pidProfiles().stream().map(this::buildDcRequest).toList();
+    publishToDataCite(dcRequests, mediaObjectEvent.eventType());
+  }
+
+  private void publishToDataCite(List<DcRequest> requests, EventType eventType) {
+    int successCount = 0;
+    for (var request : requests) {
+      var body = mapper.valueToTree(request);
+      try {
+        var method = eventType.equals(EventType.CREATE) ? HttpMethod.POST : HttpMethod.PUT;
+        var response = dataCiteClient.sendDoiRequest(body, method);
+        log.debug("received response from datacite: {}", response);
```

Comment:
Maybe a generic info log statements might be helpful. Like "Pushing PID: {}" to datacite" and "PID {} successfully published to datacite".